www/waterfall.js

```diff
@@ -158,52 +160,63 @@ function SelectRequest(step, request) {
             details += '<b>Request ID: </b>' + htmlEncode(r['request_id']) + '<br>';
         if (r['client_port'] !== undefined && r['client_port'] !== null && r['client_port'])
             details += '<b>Client Port: </b>' + htmlEncode(r['client_port']) + '<br>';
+        if (r['renderBlocking'] !== undefined) {
+            details += '<b>Render Blocking Status: </b>' + htmlEncode(r['renderBlocking']) + '<br>';
+
+            if (r['renderBlocking']=== 'blocking') {
+                experiment += '</ul><ul class="experiments"><li><a href="#" data-spof="' + htmlEncode(r['host']) + '">Run a <abbr title="Single Point of Failure">SPOF</abbr> Test (Runs in a new tab)</a></li>';
+            }
+        }
+        details += '<h3>Timing</h3>';
         if (r['created'] !== undefined)
             details += '<b>Discovered: </b>' + (r['created'] / 1000.0).toFixed(3) + ' s<br>';
         if (r['load_start'] !== undefined)
             details += '<b>Request Start: </b>' + (r['load_start'] / 1000.0).toFixed(3) + ' s<br>';
         if (IsValidDuration(r['dns_ms'])) {
             details += '<b>DNS Lookup: </b>' + htmlEncode(r['dns_ms']) + ' ms<br>';
+            totalConnectionTime += parseFloat(r['dns_ms']);

```

Comment:
Do we have a guarantee that anything we're passing into `parseFloat` is a string that can be cast to a Float? (if not, we probably want to take care of that somewhere so we don't end up with concatenating `NaN` in an error.

Reply:
Also, are we ever parsing fractions of milliseconds? if not, maybe cast to `Number` instead?

Reply:
Hmm....good call. I'll switch to Number.

Also, for validation, I'm thinking a Number.isNaN(totalConnectionTime) check at the end probably suffices, eh?